File: codechef/FUS/prime_sieve.cpp
int N = 2000001;
int prime[2000001] = {0};

void sieve()
{
    prime[0] = 1;
    prime[1] = 1;
    int i = 2,j;
    while(i*i <= N)
    {
        j = 2*i;
        if (prime[i] != 1)
        {
            while (j<N)
            {
                prime[j] = 1;
                j += i;
            }
        }
        i += 1;
    }
}

File: codechef/FUS/prime_sieve.py
N = 2000001
prime = [0]*N
prime[0] = 1
prime[1] = 1
def sieve():
    i = 2
    while i*i <= N:
        j = 2*i
        if prime[i] != 1: 
            while j < N:
                prime[j] = 1
                j += i
        i += 1
